README.md

```diff
@@ -1,7 +1,113 @@
-PyYAML
-======
+PyYAML-ft
+=========
+
+A full-featured YAML processing framework for Python with support for free-threading
+
+## Why a fork?
+
+[PEP 703](https://peps.python.org/pep-0703/) introduced free-threaded Python as a
+separate build of CPython 3.13. Thread-safety issues that might have otherwise gone
+unnoticed are now much easier to trigger because of the absence of protection from
+the GIL. Also, because the free-threaded build is ABI-incompatible, extension
+modules need to be separate, free-threaded wheels and declare support for
+it.
+
+The PyYAML maintainers
+[decided to not port PyYAML to the free-threaded build](https://github.com/yaml/pyyaml/pull/830#issuecomment-2342475334)
+before the latter, along with Cython support for it, has been tested more extensively
+in real-world applications. Our rationale with this fork is to implement support for
+the free-threaded build, so that PyYAML can be tested with it by its users, even before
+the port is merged upstream.
+
+### Differences compared with upstream
+
+- This fork uses Cython 3.1.0b1 which supports the free-threaded build, but is still in
+  beta. Its support for the free-threaded build is also still experimental.
+- `add_constructor`, `add_representer` and `add_*resolver` now all use thread-local
+  registries, so you will have to explicitly register your custom constrcutors,
+  representers and resolvers in each thread. Here's a small test showcasing that:
+
+  ```python3
+  import io
+  import threading
+  import yaml
+
+
+  class Dice:
+      def __init__(self, first, second):
+          self.first = first
+          self.second = second
+      def __repr__(self):
+          return f"Dice({self.first}, {self.second})"
+
+
+  def construct_dice(constructor, node):
+      mapping = constructor.construct_mapping(node)
+      return Dice(**mapping)
+
+
+  def load_dice():
+      yamlcode = io.StringIO("""\
+  - !dice
+    first: 1
+    second: 6
+  - !dice
+    first: 4
+    second: 4
+  """)
+      print(f"Thread {threading.current_thread().name}")
+      try:
+          objs = yaml.load(yamlcode, Loader=yaml.CLoader)
+          print(f"\t{objs=}")
+      except Exception as e:
+          print(f"\tException occurred: {e!s}")
+
+  yaml.add_constructor("!dice", construct_dice, Loader=yaml.CLoader)
+  load_dice()
+
+  t = threading.Thread(target=load_dice)
+  t.start()
+  t.join()
+  ```
+
+  Running the above script gives the following:
+
+  ```bash
+  ❯ python3.13t tmp/t.py
+  Thread MainThread
+          objs=[Dice(1, 6), Dice(4, 4)]
+  Thread Thread-1 (load_dice)
+          Exception occurred: could not determine a constructor for the tag '!dice'
+    in "<file>", line 1, column 3
+  ```
+
```

Comment:
```suggestion
If you see new errors in multithreaded programs using `yaml-ft` that work with `yaml`, you may need to add one or more calls to `yaml.add_constructor` in your thread worker function or worker initialization function.
```

Reply:
Also does the same issue apply to representers and resolvers?

Reply:
Good addition, yes!

Reply:
Maybe worth making that explicit in the note I added above.